packages/firestore/test/unit/local/local_store.test.ts

```diff
@@ -1958,6 +1958,33 @@ function genericLocalStoreTests(
       .finish();
   });
 
+  it('handles document creation time', () => {
+    return (
+      expectLocalStore()
+        .afterAllocatingQuery(query('col'))
+        .toReturnTargetId(2)
+        .after(docAddedRemoteEvent(doc('col/doc1', 12, { foo: 'bar' }, 5), [2]))
+        .toReturnChanged(doc('col/doc1', 12, { foo: 'bar' }, 5))
+        .toContain(doc('col/doc1', 12, { foo: 'bar' }, 5))
+        .after(setMutation('col/doc1', { foo: 'newBar' }))
+        .toReturnChanged(
+          doc('col/doc1', 12, { foo: 'newBar' }, 5).setHasLocalMutations()
+        )
+        .toContain(
+          doc('col/doc1', 12, { foo: 'newBar' }, 5).setHasLocalMutations()
+        )
+        .afterAcknowledgingMutation({ documentVersion: 13 })
+        // We haven't seen the remote event yet
+        .toReturnChanged(
+          doc('col/doc1', 13, { foo: 'newBar' }, 5).setHasCommittedMutations()
+        )
+        .toContain(
+          doc('col/doc1', 13, { foo: 'newBar' }, 5).setHasCommittedMutations()
+        )
+        .finish()
+    );
+  });
+
```

Comment:
I wonder what happens when:
1. we create a document with set locally.

2. that set gets accepted first, leading us to having a document with createTime=0 in the cache. This is problematic for count because it will "discount" the document. We should probably special case for SetMutation acknowledgement: if the remote doc has no createTime (or there is no createTime), we should use the document's version as createTime when we update remote document cache.

3. Then will the following remote event have a correct createTime for the document? will we save it if they do? We should be good because watch stream is separate, and it will continue to send the new document in the remote event. And we are saving createTime now. 

Let's work on this together today!